### DISCUSSION TITLE
[论文推荐] On the huge importance of non-tech roles in Open Source: Empirical study on NPM

### DISCUSSION THREAD

**Discussion by will-ww** [X-lab2017/open-wonderland#150] [3️⃣ 聊学术]
一篇不错的论文，可以考虑推荐到我们的论文库中：

- On the huge importance of non-tech roles in Open Source: Empirical study on NPM
- https://livablesoftware.com/importance-of-non-tech-contributor-roles-open-source/

This paper aims to investigate whether this is actually the case by analyzing the frequency and diversity of non-coding contributions in OSS development. Is the number of non-coding contributors indeed significant, or OSS projects are still very much “code-driven”? And, what about the contributor diversity?, i.e., are these contributors only collaborating to the project in these non-coding roles, or are they just the same group of core developers taking all types of roles depending on the needs of the project? Or, are these non-coding contributors migrating to coding roles?. These are some of the questions we have studied in the paper: [On the analysis of non-coding roles in open source development ](https://link.springer.com/article/10.1007/s10664-021-10061-x)led by [Javier Cánovas](https://jlcanovas.es/) and published in the [Empirical Software Engineering](https://www.springer.com/journal/10664) Journal.

**Comment by xgdyp** [comment]
读了下，论文里定义了几类角色，特别是把developer（参与pr贡献）reporter（参与）comment（评论的人）reactor（使用reaction的人）区分开，这点比之前我们度过的论文细很多
特别是，里面提到了一个叫SourceCred的工具，也是用作量化社区贡献的，地址是https://sourcecred.io/ ，一会跑一下试一试～

**Comment by xgdyp** [comment]
折腾了半天，用sourcecred跑了下opendigger的结果
| Description | Cred | % |
| --- | --- | --- |
| open-digger-bot | 836.7 | 46.1% |
| frank-zsy | 574.2 | 31.6% |
| sunshinemingo | 100.0 | 5.5% |
| xiaoya-yaya | 57.1 | 3.1% |
| wengzhenjie | 28.4 | 1.6% |
| gymgym1212 | 26.0 | 1.4% |
| stevending1st | 15.3 | 0.8% |
| heming6666 | 15.0 | 0.8% |
| maninhill | 13.2 | 0.7% |
| LinuxSuRen | 13.1 | 0.7% |
| yangli-os | 10.1 | 0.6% |
| github-actions | 10.0 | 0.5% |
| birdflyi | 8.9 | 0.5% |
| yoyo-wu98 | 8.6 | 0.5% |
| will-ww | 8.4 | 0.5% |
| longyanz | 8.3 | 0.5% |
| xgdyp | 7.1 | 0.4% |
| JingbenShi668 | 5.9 | 0.3% |
| Best-Eminem | 5.6 | 0.3% |
| WuShaoling | 4.2 | 0.2% |

这里应该是值配置了计算pr相关的
<img width="1199" alt="image" src="https://user-images.githubusercontent.com/37795442/182516044-389897b4-4d93-4b04-a3c9-6c35df98508b.png">

具体的使用方法见：https://github.com/sourcecred/template-instance#setup-and-usage

一会再更合理地配置一下跑一遍和我们的openrank进行下对比～

**Reply by xgdyp** [reply]
调了半天 cred值没啥变化，可能是哪里搞错了😅
